ui/src/app/guards/sign-in-guard.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  CanActivateChild,
+  Router, RouterStateSnapshot
+} from '@angular/router';
+
+import {Observable} from 'rxjs/Observable';
+
+
+import {SignInService} from 'app/services/sign-in.service';
+
+declare const gapi: any;
+
+@Injectable()
+export class SignInGuard implements CanActivate, CanActivateChild {
+  constructor(private signInService: SignInService, private router: Router) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
+    return new Observable((observer) => {
+      this.signInService.user.subscribe(user => {
+
```

Comment:
I assume that for the purposes of chunking the work this PR is intentionally ignoring anything to do with redirecting, correct?

Reply:
Correct. About to leave the house, but I’ll get to other comments when I get back.